### ISSUE TITLE
QRoute.withChild Navigation issue

### ISSUE THREAD

**Issue by ajaylg** [SchabanBo/qlevar_router#17]
Hii @SchabanBo
I referred this example https://github.com/SchabanBo/qr_samples/blob/main/lib/common_cases/dashboard.dart
and created the same structure like signup, login, and Dashboard(with child route)
I have a sidebar in my site that has sub-routes like Dashboard/info, Dashboard/contact ..etc when I click those items it navigates to the corresponding page, in my case this scenario does not work sometimes

package version: 1.1.3
issue faced:
navigate to child-1
navigate to child-2
navigate to child-3
This child1 roue,child2 route,child3 route navigations are in static fixed sidebar
then I navigate to child-1 but it navigates to child-2, child pages did not get Refresh sometimes
routing Sometimes not works in QRoute.with child

In Each child page I call API in init state. but sometimes navigations not starts from init state so i have to manually refresh whole page to recall API

**Comment by ajaylg** [comment]
@SchabanBo 

QR: Finding Match for / under root
QR: adding Route: Key: [0](/), Full Path: / to the navigator with Key: [-1](Root)
QR: setInitialRoutePath /Demo/child1
QR: Finding Match for /Demo/child1 under root
QR: adding Route: Key: [4](HR), Full Path: /Demo to the navigator with Key: [-1](Root)
QR: adding Route: Key: [5](child1), Full Path: /Demo/child1 to the navigator with Key: [48](Demo)

---then I route to child-2---
QR: Finding Match for /Demo/child2 under root
QR: adding Route: Key: [28](child2), Full Path: /Demo/child2 to the navigator with Key: [48](Demo)

---then I route to child-3---
QR: Finding Match for /Demo/child3 under root
QR: adding Route: Key: [30](child3), Full Path: /Demo/child3 to the navigator with Key: [48](Demo)
it works,

issue -  but again i route
---then I route to child-2---
QR: Finding Match for /Demo/child2 under root

---then I route to child-1---
QR: Finding Match for /Demo/child1 under root

page not reloads , it shows only cache state(child not starts from init state)

**Comment by SchabanBo** [comment]
Hi @ajaylg ,
Sorry, I couldn't answer you yesterday, I want to test it first.

So using `QR.to` will save the state of the screen and just pushes the new page on top of it, so when you pop to it again, it will not be created again.
If you don't want to save the state use `QR.navigatorOf('/dashboard').replaceAll` so the other pages get deleted when you add a new one. and then created again when you call it.

```dart
ListTile(
  title: Text('Info'), 
  onTap: () => QR.navigatorOf('/dashboard').replaceAll('/info'),
  //onTap: () => QR.to('/dashboard/info'),
),
ListTile(
  title: Text('Orders'),
  onTap: () => QR.navigatorOf('/dashboard').replaceAll('/orders'),
  //onTap: () => QR.to('/dashboard/orders'),
),
ListTile(
  title: Text('Items'),
  onTap: () => QR.navigatorOf('/dashboard').replaceAll('/items'),
  //onTap: () => QR.to('/dashboard/items'),
)
```

Please note here when call `QR.navigatorOf` you need to call the path after the navigator path.
So in the example here you need to call just `info` in the `dashboard` navigator to get to `/dashboard/info`

I noticed here a bug too, with the initRoute (the info page in our example) the key of it is wrong. and that is why you get this wired behaver when you navigate between the screen. I will try to fix it too.

I know this should be documented somewhere. The problem is the Time 😢 

I have updated the [dashboard example](https://github.com/SchabanBo/qr_samples/blob/64e59fd7aaf39855bd228bc1d333b986a941a3c4/lib/common_cases/dashboard.dart#L128) so you can pull and test it.

**Comment by ajaylg** [comment]
oh i tried this way 

> Hi @ajaylg ,
> Sorry, I couldn't answer you yesterday, I want to test it first.
> 
> So using `QR.to` will save the state of the screen and just pushes the new page on top of it, so when you pop to it again, it will not be created again.
> If you don't want to save the state use `QR.navigatorOf('/dashboard').replaceAll` so the other pages get deleted when you add a new one. and then created again when you call it.
> 
> ```dart
> ListTile(
>   title: Text('Info'), 
>   onTap: () => QR.navigatorOf('/dashboard').replaceAll('/info'),
>   //onTap: () => QR.to('/dashboard/info'),
> ),
> ListTile(
>   title: Text('Orders'),
>   onTap: () => QR.navigatorOf('/dashboard').replaceAll('/orders'),
>   //onTap: () => QR.to('/dashboard/orders'),
> ),
> ListTile(
>   title: Text('Items'),
>   onTap: () => QR.navigatorOf('/dashboard').replaceAll('/items'),
>   //onTap: () => QR.to('/dashboard/items'),
> )
> ```
> 
> Please note here when call `QR.navigatorOf` you need to call the path after the navigator path.
> So in the example here you need to call just `info` in the `dashboard` navigator to get to `/dashboard/info`
> 
> I noticed here a bug too, with the initRoute (the info page in our example) the key of it is wrong. and that is why you get this wired behaver when you navigate between the screen. I will try to fix it too.
> 
> I know this should be documented somewhere. The problem is the Time 😢
> 
> I have updated the [dashboard example](https://github.com/SchabanBo/qr_samples/blob/64e59fd7aaf39855bd228bc1d333b986a941a3c4/lib/common_cases/dashboard.dart#L128) so you can pull and test it.

@SchabanBo 
oh I tried this solution Page navigates perfectly
but URL not updated while routing

QR.navigator.replaceAll('/login');                     ---> http://localhost:62431/login
QR.navigator.replaceAll('/dashboard');            ---> http://localhost:62431/dashboard
QR.navigatorOf('dashboard').replaceAll('/child1');       ---> http://localhost:62431/dashboard/child1  **(initial route)**

when I navigate to child2 (page navigates but URL not updated)
QR.navigatorOf('dashboard').replaceAll('/child2');       ---> http://localhost:62431/dashboard/**child1**

**Comment by SchabanBo** [comment]
@ajaylg 
Oh right, I didn't notice this one.
I will fix it properly today.

**Comment by ajaylg** [comment]
Thanks @SchabanBo

**Comment by SchabanBo** [comment]
So that is fixed with 1.2.0 😀

**Comment by ajaylg** [comment]
Thanks @SchabanBo 😀
